include/storage/shared_monitor.hpp

```diff
@@ -42,7 +42,16 @@ template <typename Data> struct SharedMonitor
 
     SharedMonitor(const Data &initial_data)
     {
-        shmem = bi::shared_memory_object(bi::open_or_create, Data::name, bi::read_write);
+        try
+        {
+            shmem = bi::shared_memory_object(bi::open_or_create, Data::name, bi::read_write);
+        }
+        catch (const bi::interprocess_exception &exception)
+        {
+            auto message =
+                boost::format("could not create directory or file '%1%' \n") % exception.what();
```

Comment:
I would include the name of the memory block to make it consistent with the other exception messages below.
```suggestion
                boost::format("Unable to open or create shared memory block '%1%'") % (const char *)Data::name;
```